include/pmacc/math/vector/Vector.hpp

```diff
@@ -123,6 +123,25 @@ namespace pmacc
 
             constexpr Vector() = default;
 
+            /** Initialize via a generator expression
+             *
+             * The generator must return the value for the corresponding index of the component which is passed to the
+             * generator.
+             */
+            template<typename F, std::enable_if_t<std::is_invocable_v<F, std::integral_constant<int, 0>>, int> = 0>
+            ALPAKA_FN_HOST_ACC constexpr explicit Vector(F&& generator)
+                : Vector(std::forward<F>(generator), std::make_integer_sequence<int, dim>{})
```

Comment:
```suggestion
                : Vector(std::forward<F>(generator), std::make_index_sequence<dim>{})
```

Reply:
will change it to using `uint32_t` because `size_t` range is to large and not required.

Reply:
@ikbuibui I updated this to `uint32_t`